lib/puppet/provider/dsc_base_provider/dsc_base_provider.rb

```diff
@@ -306,7 +306,17 @@ def invoke_get_method(context, name_hash)
       end
 
       # Convert DateTime back to appropriate type
-      data[type_key] = Puppet::Pops::Time::Timestamp.parse(data[type_key]) if context.type.attributes[type_key][:mof_type] =~ /DateTime/i
+      if context.type.attributes[type_key][:mof_type] =~ /DateTime/i && !data[type_key].nil? && !data[type_key].is_a?(String)
+        context.err('Value returned for DateTime is invalid')
+        return nil
+      elsif context.type.attributes[type_key][:mof_type] =~ /DateTime/i && !data[type_key].nil?
```

Comment:
I _think_ we can drop this first if block (it's implied by the failure to parse in the begin/rescue below) and just execute this whole thing if the type_key is a DateTime and not nil.

Reply:
Have expanded the error catch to also catch this